pkg/library/libraryapplyconfiguration/apply_configuration.go

```diff
@@ -35,21 +37,41 @@ var (
 	_ AllDesiredMutationsGetter = &applyConfiguration{}
 )
 
-func ValidateAllDesiredMutationsGetter(s AllDesiredMutationsGetter) error {
+func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutationsGetter, allAllowedOutputResources *libraryoutputresources.OutputResources) error {
 	errs := []error{}
 
-	if s == nil {
+	if allDesiredMutationsGetter == nil {
 		return fmt.Errorf("applyConfiguration is required")
 	}
 
+	allMutationRequests := []manifestclient.SerializedRequestish{}
 	for _, clusterType := range sets.List(AllClusterTypes) {
-		desiredMutationsGetter := s.MutationsForClusterType(clusterType)
+		desiredMutationsGetter := allDesiredMutationsGetter.MutationsForClusterType(clusterType)
 		switch {
 		case desiredMutationsGetter == nil:
 			errs = append(errs, fmt.Errorf("mutations for %q are required even if empty", clusterType))
 		case desiredMutationsGetter.GetClusterType() != clusterType:
 			errs = append(errs, fmt.Errorf("mutations for %q reported type=%q", clusterType, desiredMutationsGetter.GetClusterType()))
 		}
+
+		allMutationRequests = append(allMutationRequests, desiredMutationsGetter.Requests().AllRequests()...)
```

Comment:
I think that `allMutationRequests` contains already filtered output. 

`AllRequests` does: https://github.com/openshift/multi-operator-manager/blob/3db370e102a9182e54ae8cf88002f575e2823b96/pkg/library/libraryapplyconfiguration/client_mutations.go#L110